receiver/datadogreceiver/tags.go

```diff
@@ -0,0 +1,142 @@
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	semconv "go.opentelemetry.io/collector/semconv/v1.16.0"
+)
+
+// See:
+// https://docs.datadoghq.com/opentelemetry/schema_semantics/semantic_mapping/
+// https://github.com/DataDog/opentelemetry-mapping-go/blob/main/pkg/otlp/attributes/attributes.go
+var datadogKnownResourceAttributes = map[string]string{
+	"env":     semconv.AttributeDeploymentEnvironment,
+	"service": semconv.AttributeServiceName,
+	"version": semconv.AttributeServiceVersion,
+
+	// Container-related attributes
+	"container_id":   semconv.AttributeContainerID,
+	"container_name": semconv.AttributeContainerName,
+	"image_name":     semconv.AttributeContainerImageName,
+	"image_tag":      semconv.AttributeContainerImageTag,
+	"runtime":        semconv.AttributeContainerRuntime,
+
+	// Cloud-related attributes
+	"cloud_provider": semconv.AttributeCloudProvider,
+	"region":         semconv.AttributeCloudRegion,
+	"zone":           semconv.AttributeCloudAvailabilityZone,
+
+	// ECS-related attributes
+	"task_family":        semconv.AttributeAWSECSTaskFamily,
+	"task_arn":           semconv.AttributeAWSECSTaskARN,
+	"ecs_cluster_name":   semconv.AttributeAWSECSClusterARN,
+	"task_version":       semconv.AttributeAWSECSTaskRevision,
+	"ecs_container_name": semconv.AttributeAWSECSContainerARN,
+
+	// K8-related attributes
+	"kube_container_name": semconv.AttributeK8SContainerName,
+	"kube_cluster_name":   semconv.AttributeK8SClusterName,
+	"kube_deployment":     semconv.AttributeK8SDeploymentName,
+	"kube_replica_set":    semconv.AttributeK8SReplicaSetName,
+	"kube_stateful_set":   semconv.AttributeK8SStatefulSetName,
+	"kube_daemon_set":     semconv.AttributeK8SDaemonSetName,
+	"kube_job":            semconv.AttributeK8SJobName,
+	"kube_cronjob":        semconv.AttributeK8SCronJobName,
+	"kube_namespace":      semconv.AttributeK8SNamespaceName,
+	"pod_name":            semconv.AttributeK8SPodName,
+
+	// Other
+	"process_id":       semconv.AttributeProcessPID,
+	"error.stacktrace": semconv.AttributeExceptionStacktrace,
+	"error.msg":        semconv.AttributeExceptionMessage,
+}
+
+// translateDatadogTagToKeyValuePair translates a Datadog tag to a key value pair
+func translateDatadogTagToKeyValuePair(tag string) (key string, value string) {
+	if tag == "" {
+		return "", ""
+	}
+
+	key, val, ok := strings.Cut(tag, ":")
+	if !ok {
+		// Datadog allows for two tag formats, one of which includes a key such as 'env',
+		// followed by a value. Datadog also supports inputTags without the key, but OTel seems
+		// to only support key:value pairs.
+		// The following is a workaround to map unnamed inputTags to key:value pairs and its subject to future
+		// changes if OTel supports unnamed inputTags in the future or if there is a better way to do this.
+		key = fmt.Sprintf("unnamed_%s", tag)
```

Comment:
is this common practice elsewhere as well? As a user that tagged something with "foo", I'd probably expect to see it translated to `foo=foo` or `foo=true`, or even just "foo=" (empty value).

Reply:
The problem doing that is that perhaps there was already a label name called `foo`? 

Because we don't know in advance, I thought about prefixing the label with something that is not common, such as `unnamed_`. 

I particularly don't make use of unnamed tags so I wonder what would be the best approach here too. We can always change it when some users try it out.

Reply:
The problem with changing it later is that we'd break users who start depending on the behavior introduced on this PR here. I don't have a good solution for the moment, and I think it would be OK to add a note to the readme stating that it currently is "unnamed_" and might change later, and that the appropriate solution is to perhaps be more explicit (foo:foo, foo:, :foo, foo:true).